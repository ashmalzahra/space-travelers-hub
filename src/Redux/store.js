import { configureStore } from '@reduxjs/toolkit';
import rocketsReducer from './RocketsSlice';
import missionsSlice from './MissionsSlice';
// import missionsReducer from "./MissionsSlice";

const store = configureStore({
  reducer: {
    rockets: rocketsReducer,
    missions: missionsSlice.reducer,
  },
});

export default store;
