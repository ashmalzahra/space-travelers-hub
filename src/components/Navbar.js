import { NavLink } from 'react-router-dom';
import Logo from '../Images/planet.png';

function Navbar() {
  return (
    <div className="NavBar">
      <div className="LogoContainer">
        <img className="planet" src={Logo} alt="planet-logo" />
        <h1>Traverlers Hub</h1>
      </div>
      <nav className="NavLinks">
        <NavLink to="/" end className={({ isActive }) => (isActive ? 'currentPage' : undefined)}>Rockets</NavLink>
        <NavLink to="/missions" end className={({ isActive }) => (isActive ? 'currentPage' : undefined)}>Missions</NavLink>
        <NavLink to="/profile" end className={({ isActive }) => (isActive ? 'currentPage' : undefined)}>My Profile</NavLink>
      </nav>
    </div>
  );
}

export default Navbar;
