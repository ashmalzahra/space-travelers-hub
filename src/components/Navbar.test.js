import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import store from '../Redux/store';
import Navbar from './Navbar';

test('Navbar renders correctly', () => {
  const configureStore = store;
  const component = renderer.create(
    <Provider store={configureStore}>
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>
    </Provider>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

// test('Missions renders correctly', () => {
//   const store = configureStore();
//   const component = renderer.create(
//     <Provider store={store}>
//       <MemoryRouter>
//         <Missions />
//       </MemoryRouter>
//     </Provider>,
//   );
//   const tree = component.toJSON();
//   expect(tree).toMatchSnapshot();
// });

// test('Rockets renders correctly', () => {
//   const store = configureStore();
//   const component = renderer.create(
//     <Provider store={store}>
//       <MemoryRouter>
//         <Rockets />
//       </MemoryRouter>
//     </Provider>,
//   );
//   const tree = component.toJSON();
//   expect(tree).toMatchSnapshot();
// });
