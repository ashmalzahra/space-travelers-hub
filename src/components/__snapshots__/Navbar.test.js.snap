// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
<div
  className="NavBar"
>
  <div
    className="LogoContainer"
  >
    <img
      alt="planet-logo"
      className="planet"
      src="planet.png"
    />
    <h1>
      Traverlers Hub
    </h1>
  </div>
  <nav
    className="NavLinks"
  >
    <a
      aria-current="page"
      className="currentPage"
      href="/"
      onClick={[Function]}
    >
      Rockets
    </a>
    <a
      href="/missions"
      onClick={[Function]}
    >
      Missions
    </a>
    <a
      href="/profile"
      onClick={[Function]}
    >
      My Profile
    </a>
  </nav>
</div>
`;
